src/commands/hop.tsx

```diff
@@ -23,29 +28,45 @@ function Hop() {
         void getLocalBranches();
     }, [git, setAllBranches, setCurrentBranch]);
 
-    const handleSelect = (item: { label: string; value: string }) => {
-        const updateCurrentBranch = async () => {
-            const { current } = await git.branchLocal();
-            setNewBranch(current);
-        };
+    const handleSelect = useCallback(
+        (item: { label: string; value: string }) => {
+            const updateCurrentBranch = async () => {
+                const { current } = await git.branchLocal();
+                setNewBranch(current);
+            };
 
-        git.checkout(item.value)
-            .then(async () => {
-                return updateCurrentBranch();
-            })
-            .catch((error: Error) => {
-                setError(error);
-            })
-            .finally(() => {
-                exit();
-            });
-    };
+            git.checkout(item.value)
+                .then(() => {
+                    void updateCurrentBranch();
+                })
+                .catch((error: Error) => {
+                    setError(error);
+                })
+                .finally(() => {
+                    exit();
+                });
+        },
+        [git, setNewBranch, setError, exit]
+    );
 
     const items = useMemo(() => {
+        if (!currentBranch) return [];
+
         if (allBranches.length === 1) {
             return [{ label: currentBranch, value: currentBranch }];
         }
-        return allBranches
+
+        const searchedBranches = allBranches.filter((branch) =>
+            maybeSearchTerm
+                ? branch.toLowerCase().includes(maybeSearchTerm.toLowerCase())
+                : true
+        );
+
+        if (!searchedBranches.length) {
+            return [];
+        }
```

Comment:
If searched branches has no length then the following filter and map will still return an empty array

Reply:
If that's the case then the previous filter and the following filter could be combined to save on iterations

Reply:
I noticed as you were reviewing, I believe an earlier commit resolves this